typescript/utils/src/types.ts

```diff
@@ -2,6 +2,7 @@ import type { SignatureLike } from '@ethersproject/bytes';
 import type { BigNumber, ethers } from 'ethers';
 
 export enum ProtocolType {
+  ZKSync = 'zksync',
```

Comment:
instead of adding a `zksync` protocol type, I think you should add to the `ChainTechnicalStack` enum. We already have an entry there for arbitrum nitro chains https://github.com/hyperlane-xyz/hyperlane-monorepo-zksync/blob/0891c62339d3a65f60bb890682279e0d2fcf30cd/typescript/sdk/src/metadata/chainMetadataTypes.ts#L19

Reply:
this will help minimise the diff a bit, by reducing how often you need to reimplement ethereum functions/calls for zksync

Reply:
agree, love it

Reply:
think I've changed my mind now. Think it does make sense to have it as a ProtocolType instead of a ChainTechnicalStack, seeing the intricacies of using ethers VS zksync-ethers providers/signers